jscomp/test/VariantCoercion.res

```diff
@@ -30,3 +30,17 @@ module CoerceVariants = {
   let x: x = One({age: 1})
   let y: y = (x :> y)
 }
+
+module CoerceWithPayload = {
+  @unboxed type strings = String(string) | First | Second | Third
+  let a: strings = String("hello")
+  let aa: strings = First
+  let b: string = (a :> string)
```

Comment:
These 2 lines test the same thing. Does not hurt, but it's not helpful either.
Coercion works on types and does not depend on the specific instance.

Reply:
That said, it's kind of nice as documentation of the intentions.

Reply:
Yeah that was my intention, to illustrate that it works for both the catch all case and the regular cases. But yes, it's not actually useful as a test, just illustrating for documentation purposes.